### ISSUE TITLE
Split code that's not generated in Hercules.Database

### ISSUE THREAD

**Issue by domenkozar** [expipiplus1/hercules#1]
For example:

```
instance ElmType Project where
```

Or check if upstream provides a way to add extra instances for each type.

**Comment by domenkozar** [comment]
@expipiplus1 do you think we could use something like https://github.com/ndmitchell/derive to separate `deriving` things?

**Comment by expipiplus1** [comment]
I think all that we need to derive can be done with template Haskell or ghc-mod at the moment. Was their anything you had in mind?

**Comment by domenkozar** [comment]
Yes, deriving `Generic`, `ToJSON` and others could be done with template Haskell. I'm not sure what other changes we did, but we can diff :)

**Comment by expipiplus1** [comment]
Oh, do you mean the deriving in the generated Database files?

**Comment by domenkozar** [comment]
Yes, since `opaleye-gen` does add `Generic` instances, we'd have to derive them in separate file. Seems like https://github.com/ndmitchell/derive can do that for use using TH.

**Comment by expipiplus1** [comment]
the documentation for opaleye-gen recommends using it as a starting point for further modification, and not as a tool to be run on every db update. Just putting that here, we don't have to do it that way.
